src/Extensions/CMSPageEditControllerExtension.php

```diff
@@ -0,0 +1,114 @@
+<?php
```

Comment:
This class contains a bunch of code that used to be in silverstripe/cms https://github.com/silverstripe/silverstripe-cms/pull/3027